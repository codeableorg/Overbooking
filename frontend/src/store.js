import { createStore, applyMiddleware, compose } from "redux";
import reducer from "./reducer";
import logger from "redux-logger";
import thunk from "redux-thunk";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composer(applyMiddleware(thunk, logger)));

export default store;
