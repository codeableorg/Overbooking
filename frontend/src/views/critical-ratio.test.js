import React from "react";
import { render } from "@testing-library/react";
import CriticalRatio from "./critical-ratio";
import { Provider } from "react-redux";
import store from "../store";

test("CriticalRatio component", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <CriticalRatio />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
