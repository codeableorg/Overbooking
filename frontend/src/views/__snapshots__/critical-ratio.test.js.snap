// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriticalRatio component 1`] = `
<DocumentFragment>
  <form
    class="css-1gf3fd4-CriticalRatio"
  >
    <img
      alt="plane"
      class="css-1d6aglg-CriticalRatio"
      src="plane.png"
    />
    <div
      class="css-1pgmn93-CriticalRatio"
    >
      Flight PDR456
    </div>
    <div
      class="css-bgusvs-CriticalRatio"
    >
      <div
        class="css-miwez1-CriticalRatio"
      >
        Cost of Overbooking
        <p>
          600
        </p>
      </div>
      <div
        class="css-miwez1-CriticalRatio"
      >
        Underage Cost
        <p>
          300
        </p>
      </div>
    </div>
    <div
      class="css-1pgmn93-CriticalRatio"
    >
      Calculate the Critical Ratio
    </div>
    <input
      aria-label="enter critical ratio"
      autocomplete="off"
      class="css-7j0mhv-CriticalRatio"
      id="critical-ratio"
      name="critical-ratio"
      placeholder="enter the critical ratio"
      required=""
      type="number"
      value=""
    />
    <button
      class="css-1kbyqv2-CriticalRatio"
      disabled=""
      type="submit"
    >
      Set Critical Ratio
    </button>
  </form>
</DocumentFragment>
`;
